### PR TITLE
Pds1 comms 519

### LINKED ISSUE #519: Implement new comms system on PDS1 code
Literally copy-pasted from #505 
The new comms system should be implemented in the PDS1 code. All the commands should be documented

### PR DESCRIPTION
**Pull request by MewtR**
# Assignee Section

## Description
This PR makes the old PDS use the new comms library.

* CMakelists.txt that is used to compile the new PDS and link it against the internal comms library.
* Old PDS commands separated into their own functions.

### Steps for Testing
Note: You need an ATmega328p to run the tests. I used and Arduino Nano.

1. Use the CMakeLists.txt provided in `robot/rover/PDS` to compile the old pds.
2. Upload it to your device.
3. Make sure you have a way to view your devices output. For example `tail -f` technique from the sending bytes section.
4. Send raw bytes corresponding to a command. Refer to [here](https://docs.google.com/spreadsheets/d/1bE3h0ZCqPAUhW6Gn6G0fKEoOPdopGTZnmmWK1VuVurI/edit#gid=1682378273) for available commands
5. You should printed statements corresponding to the command you issued.

#### Additional info 
copied from [here](https://github.com/space-concordia-robotics/robotics-prototype/pull/516)

It's not trivial to use the Arduino IDE for the testing steps above so I had to do everything from the commandline. Here's how:

###### Compiling

Use the provided `CMakelists.txt`. These generate elf files

###### Uploading
The arduino ide will only upload a binary it compiled itself so we need to manually upload with `avrdude`. My command looked like the following `avrdude -C/home/bionic/arduino-1.8.12/hardware/tools/avr/etc/avrdude.conf -v -patmega328p -carduino -P/dev/ttyUSB0 -b57600 -D -Uflash:w:bin/PDS2.elf:e`. Just remember to change the device from `/dev/ttyUSB0` to whatever your device is and the path to `avrdude.conf` to the path on your system. Also put the path to your elf file in the last part : `-Uflash:w:<path/to/elf.elf>:e`

##### Sending bytes
The Arduino IDE's serial monitor only allows you to send ascii as far as I can tell. So I used python to send in raw bytes.
Basically in one terminal you should have `tail -f <device>` so for example `tail -f /dev/ttyUSB0`.
In another terminal you can run something like: `python3 -c "import sys; sys.stdout.buffer.write(b'\x36\x00\x02\x00\x02\x06\x64\x00')" > /dev/ttyUSB0` and you should see any results in the first terminal. If you don't see anything despite having some prints. Try sending data via the Arduino IDE's serial monitor first and then retry this method.

closes #519 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [ ] Local Test Performed Successfully
- [ ] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

File: robot/rover/PDS/PDS.ino [hunk 1]
```diff
@@ -152,16 +161,7 @@ void loop() {
...
-    if (num > 0) {
-      Serial.println(BUFFER); // echo the received command
-      parseCommand(); // check the command and (in some cases) react to it
-      memset(BUFFER, 0, CHAR_BUFF_SIZE);
-      messagesReceived++;
-    } else {
-      Serial.println("Command error: PDS received empty message");
-      badMessages++;
-    }
+    internal_comms::readCommand(commandCenter);
```
Comment (MartensCedric) [inline robot/rover/PDS/PDS.ino:164]: HOT

File: robot/rover/PDS/PDS.ino [hunk 2]
```diff
@@ -112,8 +115,14 @@ void printMotorState();
...
 float readMultiplexer(uint8_t channel);
 
+const uint8_t TX_PIN = 1;
+const uint8_t RX_PIN = 0;
+
+internal_comms::CommandCenter* commandCenter = new PDSCommandCenter();
+
 void setup() {
-  Serial.begin(SERIAL_BAUD_RATE);
+  //Serial.begin(SERIAL_BAUD_RATE);
```
Comment (MartensCedric) [inline robot/rover/PDS/PDS.ino:124]: pls remove

File: robot/rover/PDS/PDSCommandCenter.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        resetCurrentReadingErrorFlags();
+    }else if(commandID == COMMAND_TOGGLE_AUTO_MODE) 
+    {
+        uint8_t state = *rawArgs;
+        toggleAutoProtection(state);
+    }
+    else
+    {
+            error();
+    }
```
Comment (MartensCedric) [inline robot/rover/PDS/PDSCommandCenter.cpp:59]: is it me or the indentation is weird?  looks like there's 8 space shit going on or something

**Review by MartensCedric — CHANGES_REQUESTED** [review]
Looks good. Pretty busy with midterms rn but Ill make a test soon. Also tim is working on the send command so we will be able to test again

File: robot/rover/PDS/PDSCommandCenter.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            motor(motorPin, state);
+    }else if(commandID == COMMAND_FAN)
+    {
+            if (rawArgsLength != 2) {
+                Serial.println("Wrong number of arguments. It should two, fan number + fan speed");
+                return;
+            }
+            uint8_t fanNum = *rawArgs;
+            uint8_t fanSpeed = *(++rawArgs);
+            fan(fanNum, fanSpeed);
```
Comment (VimFreeman) [inline robot/rover/PDS/PDSCommandCenter.cpp:44]: 8 indent monkaS

**Comment by VimFreeman** [comment]
Closing bc we are working on PDS2 and not using PDS1

**Pull request closed without merging**